File: Typescript/src/readBinaryWatch.ts
export function readBinaryWatch(turnedOn: number): string[] {
	const result: string[] = [];

	const countBits = (n: number): number => {
		let count = 0;
		while (n > 0) {
			count += n & 1;
			n >>= 1;
		}
		return count;
	};

	for (let hour = 0; hour < 12; hour++) {
		for (let minute = 0; minute < 60; minute++) {
			if (countBits(hour) + countBits(minute) === turnedOn) {
				const timeString = `${hour}:${minute.toString().padStart(2, '0')}`;
				result.push(timeString);
			}
		}
	}

	return result;
};

File: Dart/lib/readBinaryWatch.dart
class Solution {
  List<String> readBinaryWatch(int turnedOn) {
    var result = <String>[];
    for (var h = 0; h < 12; h++) {
      for (var m = 0; m < 60; m++) {
        if (countBit(h) + countBit(m) == turnedOn) {
          result.add('$h:${m.toString().padLeft(2, '0')}');
        }
      }
    }
    return result;
  }

  int countBit(int n) {
    var count = 0;
    while (n != 0) {
      n &= (n - 1);
      count++;
    }
    return count;
  }
}
